### PR TITLE
Update focalboard to be compatible with mattermost v6

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR updates `focalboard` to work with the `mattermost v6`. It's currently marked as a draft waiting for the v6 update PR to be merged.

### CODE REVIEW

File: mattermost-plugin/go.mod [hunk 1]
```diff
@@ -4,10 +4,12 @@ go 1.12
 
 replace github.com/mattermost/focalboard/server => ../server
 
+replace github.com/mattermost/mattermost-plugin-api/v6 => ../../../../../../dev/mattermost-plugin-api
```
Comment (mgdelacroix) [inline mattermost-plugin/go.mod:7]: This `replace` exists for testing purposes. Once the `mattermost-plugin-api` v6 PR gets merged, the dependency will be updated and the `replace` removed.

File: mattermost-plugin/plugin.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "release_notes_url": "https://github.com/mattermost/focalboard/releases",
     "icon_path": "assets/starter-template-icon.svg",
     "version": "0.9.0",
-    "min_server_version": "5.37.0",
+    "min_server_version": "5.38.0",
```
Comment (mgdelacroix) [inline mattermost-plugin/plugin.json:10]: We cannot depend yet on `"6.0.0"` as the version does not exist in `mattermost-server`

**Comment by jespino** [comment]
I like this, whenever we can fix the plugin-api problem, I think we can merge it.

File: mattermost-plugin/plugin.json (hunk 2 above)
Comment (chenilim) [reply mattermost-plugin/plugin.json:10]: Yup, that's dependent on mattermost-server [PR-18247](https://github.com/mattermost/mattermost-server/pull/18247). We should update our dependency to v6.0 once that change is rolled out to community and community-daily, which should happen on Aug 25 ([thread](https://community.mattermost.com/core/pl/hrq9g5m6y7d75gftaxyp7iiu3y)).

**Review by jespino — APPROVED** [review]

**Pull request merged**